src/aura/BGE_DataImportBatchEntry/BGE_DataImportBatchEntryHelper.js

```diff
@@ -0,0 +1,241 @@
+({
+    /**
+     * @description: creates the form component
+     */
+    createEntryForm: function (component) {
+        $A.createComponent(
+            "c:BGE_EntryForm",
+            {
+                "aura:id": "entryForm",
+                "labels": component.get("v.labels"),
+                "donorType": component.get("v.donorType"),
+                "recordId": component.get("v.recordId"),
+                "dataImportFields": component.get("v.dataImportFields")
+            },
+            function(newComponent, status, errorMessage){
+                //Add the new component to the body array
+                if (status === "SUCCESS") {
+                    var body = component.get("v.body");
+                    body.push(newComponent);
+                    component.set("v.body", body);
+                }
+                else if (status === "INCOMPLETE") {
+                    console.log("No response from server or client is offline.")
+                    // Show offline error
+                }
+                else if (status === "ERROR") {
+                    console.log("Error: " + errorMessage);
+                    // Show error message
+                }
+            }
+        );
+    },
+
+    /**
+     * @description: retrieves the dataImportRows and sets them to the table.
+     */
+    getDIs: function (component) {
+        this.showSpinner(component);
+        var action = component.get("c.getDataImports");
+        action.setParams({batchId: component.get("v.recordId")});
+        action.setCallback(this, function (response) {
+            var state = response.getState();
+            if (state === "SUCCESS") {
+                var responseRows = response.getReturnValue();
+                this.setDataTableRows(component, responseRows);
+                this.setTotals(component, responseRows);
+                var openRoad = component.find("openRoadIllustration");
+                if (responseRows.length === 0) {
+                    $A.util.removeClass(openRoad, "slds-hide");
+                } else {
+                    $A.util.addClass(openRoad, "slds-hide");
+                }
+            } else {
+                this.showToast(component, $A.get("$Label.c.PageMessagesError"), response.getReturnValue(), 'error');
+            }
+            this.hideSpinner(component);
+        });
+        $A.enqueueAction(action);
+    },
+
+    /**
+     * @description: retrieves the model information. If successful, sets the model and creates child component; otherwise alerts user.
+     */
+    getModel: function(component) {
+        var action = component.get("c.getDataImportModel");
+        action.setParams({batchId: component.get("v.recordId")});
+        action.setCallback(this, function (response) {
+            var state = response.getState();
+            if (state === "SUCCESS") {
+                var response = JSON.parse(response.getReturnValue());
```

Comment:
Why do we need to parse here? (I suspect there's a good reason, just curious to hear about it. :) )

Reply:
Indeed! We need to parse the serialized string passed from Apex in order to make the data usable, and we use the string serializing/deserializing since we're passing back our own custom data model and were told not to use custom classes in lightning component attributes.